lib/shoppingcart.rb

```diff
@@ -41,20 +38,19 @@ def products_by_category(category)
      end
   end
 
-
   def percentage_occupied
-   (((total_number_of_products.to_f / capacity))*100).round(2)
+   ((total_number_of_products.to_f / capacity.to_f) * 100).round(2)
   end
 
   def sorted_products_by_quantity
     @products.sort_by do |product|
-      -product.quantity
-    end
+      product.quantity
+    end.reverse
   end
 
-  # def product_breakdown
-  #   @products.sort_by do |product|
-  #     product
-  #   end
-  # end
+  def product_breakdown
+    @products.group_by do |product|
+      product.category
+    end.sort.to_h
+  end
```

Comment:
Since `group_by` returns a hash this is a great time to use it! The example wanted it sorted alphabetically so I've taken the result and sorted it, but that messed with the hash, so I turned it back into a hash. This may not be the best way, but it works 🤷‍♀️